shell/platform/android/android_context_gl.cc

```diff
@@ -112,43 +118,86 @@ static bool TeardownSurface(EGLDisplay display, EGLSurface surface) {
 }
 
 // For onscreen rendering.
-static EGLResult<EGLSurface> CreateWindowSurface(
-    EGLDisplay display,
-    EGLConfig config,
-    AndroidNativeWindow::Handle window_handle) {
+bool AndroidContextGL::CreateWindowSurface(
+    ftl::RefPtr<AndroidNativeWindow> window,
+    bool* srgb_support) {
   // The configurations are only required when dealing with extensions or VG.
   // We do neither.
+
+  window_ = std::move(window);
+  EGLDisplay display = environment_->Display();
+
+  EGLint srgb_attributes[] = {
+      EGL_GL_COLORSPACE_KHR, EGL_GL_COLORSPACE_SRGB_KHR,
```

Comment:
When depending on GL or EGL extensions, I think the safest approach is to check if that extension is supported in the list returned from [`eglQueryString`](https://www.khronos.org/registry/EGL/sdk/docs/man/html/eglQueryString.xhtml) for the `EGL_EXTENSIONS` key.

I am not sure how EGL implementations handle unsupported keys. I hope they don't ignore unsupported keys. In that case, you will still get a valid surface but without the support you expect.